checks.d/dns_check.py

```diff
@@ -51,12 +51,14 @@ def query_dns(self, instance, timeout):
         if nameserver is not None:
             resolver.nameservers = [nameserver]
 
+        record_type = instance.get('record_type', 'A')
+
         status = AgentCheck.CRITICAL
         start_time = time.time()
         try:
             self.log.debug('Resolving hostname %s...' % hostname)
-            answer = resolver.query(hostname)
-            assert(answer.rrset.items[0].address)
+            answer = resolver.query(hostname, rdtype=record_type)
+            assert(answer.rrset.items[0].to_text())
             end_time = time.time()
         except dns.exception.Timeout:
```

Comment:
we should also catch `NoAnswer` in case the query fails for a record type that doesn't exist for the address